source/connect-array.ts

```diff
@@ -25,22 +25,23 @@ import {
   ControlContainer,
   ControlValueAccessor,
 } from '@angular/forms';
+
+// This doesn't exist in Angular 4.
 import {
-  composeAsyncValidators,
-  composeValidators,
   controlPath,
   setUpFormContainer,
   selectValueAccessor,
 } from '@angular/forms/src/directives/shared';
```

Comment:
We should not be digging around in angular's internals - we'll get burned every time they update. This constructs are not to my knowledge part of the interface covered by their semantic versioning.